src/Microsoft.ML/Models/OnnxConverter.cs

```diff
@@ -10,9 +10,50 @@ namespace Microsoft.ML.Models
     public sealed partial class OnnxConverter
     {
         /// <summary>
-        /// Converts the model to ONNX format.
+        ///
+        /// This API converts the model to ONNX format by inspecting the transform pipeline 
+        /// from the end, checking for components that know how to save themselves as ONNX. 
+        /// The first item in the transform pipeline that does not know how to save itself 
+        /// as ONNX, is considered the "input" to the ONNX pipeline. (Ideally this would be the 
+        /// original loader itself, but this may not be possible if the user used unsavable 
+        /// transforms in defining the pipe.) All the columns in the source that are a type the 
+        /// ONNX knows how to deal with will be tracked. Intermediate transformations of the 
+        /// data appearing as new columns will appear in the output block of the ONNX, with names 
+        /// derived from the corresponding column names. The ONNX json will be serialized to a 
+        /// path defined through the Json option.
+        ///
+        /// This API supports the following arguments:
+        /// <see cref="Onnx"/> indicates the file to write the ONNX protocol buffer file to. This is optional.
```

Comment:
At least one of the Onnx or Json file is required, right? From the description, I feel both are optional.

Reply:
Both are optional, if you don't specify Onnx path then it won't save the model anywhere. 

if (_outputModelPath != null)
            {
                using (var file = Host.CreateOutputFile(_outputModelPath))
                using (var stream = file.CreateWriteStream())
                    model.WriteTo(stream);
            }

            if (_outputJsonModelPath != null)
            {
                using (var file = Host.CreateOutputFile(_outputJsonModelPath))
                using (var stream = file.CreateWriteStream())
                using (var writer = new StreamWriter(stream))
                {
                    var parsedJson = JsonConvert.DeserializeObject(model.ToString());
                    writer.Write(JsonConvert.SerializeObject(parsedJson, Formatting.Indented));
                }
            }

Reply:
That's fairly peculiar is it not? Is there a usecase that justifies that strange behavior?

Reply:
I have opened issue #423 against this.